analyses/cell-type-ewings/scripts/clustering-workflow/01-clustering.R

```diff
@@ -0,0 +1,166 @@
+})
+
+option_list <- list(
+  make_option(
+    opt_str = c("--sce_file"),
+    type = "character",
+    help = "Path to processed SCE file to use for calculating clusters."
+  ),
+  make_option(
+    opt_str = c("--output_file"),
+    type = "character",
+    help = "Path to TSV file to save clustering results."
+  ),
+  make_option(
+    opt_str = c("--louvain"),
+    action = "store_true",
+    default = FALSE,
+    help = "Indicate whether or not to perform Louvain clustering."
+  ), 
+  make_option(
+    opt_str = c("--leiden_mod"),
+    action = "store_true",
+    default = FALSE,
+    help = "Indicate whether or not to perform Leiden clustering with the modularity objective function."
+  ), 
+  make_option(
+    opt_str = c("--leiden_cpm"),
+    action = "store_true",
+    default = FALSE,
+    help = "Indicate whether or not to perform Leiden clustering with the CPM objective function."
+  ), 
+  make_option(
+    opt_str = c("--nn_range"),
+    default = "5,10,15,20,25,30,35,40",
+    type = "character", 
+    help = "Comma separated list with a range of values to use for nearest neighbors parameter."
+  ), 
+  make_option(
+    opt_str = c("--louvain_res_range"),
+    default = ".5,1,1.5",
+    type = "character",
+    help = "Comma separated list with a range of values to use for resolution with the Louvain clustering algorithm.
+      Only used with the `--louvain` flag."
+  ),
+  make_option(
+    opt_str = c("--mod_res_range"),
+    default = ".5,1,1.5",
+    type = "character",
+    help = "Comma separated list with a range of values to use for resolution with the Leiden clustering algorithm and modularity objective function.
+      Only used with the `--leiden_mod` flag."
+  ),
+  make_option(
+    opt_str = c("--cpm_res_range"),
+    default = "0.001,0.005,0.01",
+    type = "character",
+    help = "Range of values to use for resolution with the Leiden clustering algorithm and CPM objective function.
+      Only used with the `--louvain` flag."
+  ),
+  make_option(
+    opt_str = c("-t", "--threads"),
+    type = "integer",
+    default = 4,
+    help = "Number of multiprocessing threads to use."
+  ),
+  make_option(
+    opt_str = c("--seed"),
+    type = "integer",
+    default = 2024,
+    help = "A random seed for reproducibility."
+  )
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# Set up -----------------------------------------------------------------------
+
+# set seed
+set.seed(opt$seed)
+
+# make sure inputs exist
+stopifnot(
+  "SCE file does not exist" = file.exists(opt$sce_file),
+  "Must indicate at least one clustering algorithm to use, one of `--louvain`, `--leiden_mod`, or `--leiden_cpm`" = opt$louvain && opt$leiden_cpm && opt$leiden_cpm
+)
+
+# create output directory if it doesn't exist
+output_dir <- dirname(opt$output_file)
+fs::dir_create(output_dir)
+
+
+# read in input sce files
+sce <- readr::read_rds(opt$sce_file)
+
+# get nn and resolution as numbers
```

Comment:
A little helper function couldn't hurt, but also I am partial to little helper functions and it's not totally necessary here. Up to you!

Either way, protip: `str_split_1` is a very helpful variant which returns a vector, not a list!

```suggestion
# get nn and resolution as numbers
split_numbers <- function(param) {
  param |>
    string::str_split_1(param, ",") |>
    as.numeric()
 }
```